### PR TITLE
Update hunspell to 1.6.1

### PR DESCRIPTION
**Pull request by stloeffler**
Development has moved from Sourceforge to GitHub.

### CODE REVIEW

File: src/hunspell.mk [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
 $(PKG)_DESCR    := Hunspell
 $(PKG)_IGNORE   :=
-$(PKG)_VERSION  := 1.3.3
-$(PKG)_CHECKSUM := a7b2c0de0e2ce17426821dc1ac8eb115029959b3ada9d80a81739fa19373246c
+$(PKG)_VERSION  := 1.6.1
+$(PKG)_CHECKSUM := 30f593733c50b794016bb03d31fd2a2071e4610c6fa4708e33edad2335102c49
 $(PKG)_SUBDIR   := hunspell-$($(PKG)_VERSION)
 $(PKG)_FILE     := hunspell-$($(PKG)_VERSION).tar.gz
-$(PKG)_URL      := https://$(SOURCEFORGE_MIRROR)/project/hunspell/Hunspell/$($(PKG)_VERSION)/$($(PKG)_FILE)
+$(PKG)_URL      := https://github.com/hunspell/hunspell/archive/v$($(PKG)_VERSION).tar.gz
```
Comment (starius) [inline src/hunspell.mk]: Can `$(PKG)_SUBDIR`, `$(PKG)_FILE`, `$(PKG)_URL` and `$(PKG)_UPDATE` be replaced with `$(PKG)_GH_CONF`?

File: src/hunspell.mk [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
-    $(SED) -n 's,.*/\([0-9][^"]*\)/".*,\1,p' | \
+    $(WGET) -q -O- 'https://github.com/hunspell/hunspell/releases' | \
+    $(SED) -n 's,.*/archive/v\([0-9][^>]*\)\.tar.*,\1,p' | \
     head -1
 endef
 
 define $(PKG)_BUILD
     # Note: the configure file doesn't pick up pdcurses, so "ui" is disabled
-    cd '$(1)' && ./configure \
+    cd '$(1)' && autoreconf -vfi && ./configure \
```
Comment (starius) [inline src/hunspell.mk]: Optional: can you build it out-of-tree? (Grep by `BUILD_DIR` and `SOURCE_DIR` for examples.)

File: src/hunspell.mk (hunk 1 above)
Comment (stloeffler) [inline src/hunspell.mk]: Done

File: src/hunspell.mk (hunk 2 above)
Comment (stloeffler) [inline src/hunspell.mk]: Done

**Comment by starius** [comment]
The build was tested on all 4 targets. Everything builds fine.
The list of files installed by the package: https://gist.github.com/62deef0bf82e42d001f72d047cd36acf

The package installs files which are not used in MXE: docs, shell scripts (e.g. wordlist2hunspell). I replaced `bin_PROGRAMS= sbin_PROGRAMS= noinst_PROGRAMS=` with `$(MXE_DISABLE_CRUFT)` which is supposed to disable installation of docs and binaries. It helped with docs but those scripts are still installed. The mk files looks like this after my changes: https://gist.github.com/f02e15c4c826d3229ea68452a00811ce

The package uses var `dist_bin_SCRIPTS` to list scripts to install. Adding it to `MXE_DISABLE_PROGRAMS` helped. I am sending this change as a [separate pull request](https://github.com/mxe/mxe/pull/1755).

File: src/hunspell.mk [hunk 3]
```diff
@@ -1,30 +1,23 @@
 # This file is part of MXE. See LICENSE.md for licensing information.
 
 PKG             := hunspell
-$(PKG)_WEBSITE  := https://hunspell.sourceforge.io/
+$(PKG)_WEBSITE  := http://hunspell.github.io/
```
Comment (starius) [inline src/hunspell.mk]: Please use https scheme when possible.

Comment (stloeffler) [inline src/hunspell.mk]: Done

**Comment by starius** [comment]
Thank you for the update!
I also used MXE_DISABLE_CRUFT and -j 1 in a separate commit and rebased the PR.

**Pull request merged**